src/actions/github.ts

```diff
@@ -0,0 +1,63 @@
+export const parseOAuthCode = () => {
+    const windowUrl = window.location.search;
+    const params = new URLSearchParams(windowUrl);
+
+    const code = params.get("code");
+
+    return code
+}
+
+const callGatekeeper = (code: string) => {
+    const url = `http://localhost:9999/authenticate/${code}`;
+
+    if (code === "") {
+        return ""
+    }
+    
+    return fetch(url)
+        .then(response => response.json())
+        .then(function(response) {
+            if (response.error) {
+                console.log(response.error);
+

```

Comment:
i like using async/await instead and try / catch for errors, looks much cleaner

Reply:
I'm definitely struggling to get the async clean. I think we backlog this as part of a later refactor in the name of progress. Let's at least get the *hack* done for the hackathon. I did give it an attempt in this next push, but it definitely smells off.

Lmk your thoughts.

Reply:
thats good for me, backlogs can be quick patches later